test/test_insert-markdown.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BASENAME=$(basename "$0")
+CANONICAL_SCRIPT=$(readlink -e "$0")
+SCRIPT_DIR=$(dirname "${CANONICAL_SCRIPT}")
+ROOT_DIR=$(dirname "${SCRIPT_DIR}")
+
+function test_simple_awk_replace {
+  GITHUB_REPOSITORY="mermaid-js/mermaid"
+  GITHUB_SHA="74d09b5468f7a6a7ba4be9deae8f34e62edbf583"
+  insert_markdown_awk="${ROOT_DIR}/insert-markdown.awk"
+  image_path="/img/gray-flow.png"
+  awk -v n="1" \
+      -v path="${image_path}" \
+      -v codeblock_action="replace" \
+      -v GITHUB_REPOSITORY="${GITHUB_REPOSITORY}" \
+      -v GITHUB_SHA="${GITHUB_SHA}" \
+      -f "${insert_markdown_awk}" \
+      "${SCRIPT_DIR}/simple.md" > "/tmp/simple.md.actual"
+  diff "${SCRIPT_DIR}/simple.md.expected" "/tmp/simple.md.actual"
+}
+
+test_simple_awk_replace
```

Comment:
This is good to have a repeatable test -- great work :tada: 

I am concerned that it doesn't really test what the workflow does or how it is currently set up.

The workflow, as it is now, take the last commit, does a diff to the one before looking for file changes.  Then based on markdown (mermaid or markdown with mermaid blocks) it compiles them with the bash script making changes.

Now correct me if I'm wrong, but how do we get the SHA of a commit that hasn't happened yet to create the image links in the awk script.  The SHA hash reference is to the commit before the images were created, and even doing a `commit --amend` would then change the SHA hash.

I'm not sure we can use the SHA reference when building the image link unless we were to build the images, commit them, get the SHA of that commit, change the markdown, then commit that.

Reply:
> I am concerned that it doesn't really test what the workflow does or how it is currently set up to be used as a github action.

This is just a typical unit test. Feel free to add integration testing afterwards ;)

> Now correct me if I'm wrong,

See https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables

JIST is, `GITHUB_SHA` and similar are provided by GitHub Action runner.